src/Refinery/DeriveApplyToFromTransform.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/* Copyright (c) 1998-2019 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * @author  Niels Theen <ntheen@databay.de>
+ */
+
+namespace ILIAS\In\Transformation;
+
+use ILIAS\Data\Result;
+
+trait DeriveApplyToFromTransform
+{
+	/**
+	 * @param Result $result
+	 * @return Result
+	 */
+	public function applyTo(Result $result) : Result
+	{
+		try {
+			$value = $this->transform($result->value());
+		} catch (\Exception $exception) {
+			return new Result\Error($exception);
+		} catch (\Error $error) {
```

Comment:
Do we really want to catch `Errors` here? This would mask every "real" error in the code-path beneath `transform`. We would only get the message, not the trace, so it would also be hard to find the problem.